stratum/hal/lib/phal/phal_cli.cc

```diff
@@ -0,0 +1,401 @@
+    grpc::ClientContext context;
+
+    // Grab the polling interval
+    std::string str;
+    int polling_interval = 5;
+    while (true) {
+        std::cout << "Polling interval(secs) 5: ";
+        std::getline(std::cin, str);
+        if (std::cin.eof() || str == "") break;
+        polling_interval = stoi(str);
+        if (polling_interval > 0) break;
+        std::cout << "Invalid polling interval: " << str << std::endl;
+    }
+
+    // Grab the of responses (0 is infinite)
+    int num_responses = 10;
+    while (true) {
+        std::cout << "Num responses(0 = infinite) 10: ";
+        std::getline(std::cin, str);
+        if (std::cin.eof() || str == "") break;
+        num_responses = stoi(str);
+        if (num_responses >= 0) break;
+        std::cout << "Invalid num responses: " << str << std::endl;
+    }
+
+    req.set_str(query);
+    req.set_polling_interval(polling_interval);
+
+    absl::Time start_time = absl::Now();
+    std::unique_ptr<grpc::ClientReader<SubscribeResponse>> reader(
+        phaldb_svc_->Subscribe(&context, req));
+
+    // Read the stream of responses (Note: return after 10)
+    int cnt = num_responses;
+    while (reader->Read(&resp)) {
+        auto result_str = resp.DebugString();
+        if (result_str.size() <= 0) {
+            std::cout << "No Results" << std::endl;
+        } else {
+            std::cout << result_str << std::endl;
+        }
+        int resp_duration = (absl::Now() - start_time) /
+                                absl::Microseconds(1);
+        std::cout << "Response in " << resp_duration << " us."
+              << std::endl;
+        start_time = absl::Now();
+
+        if (num_responses > 0 && --cnt == 0) {
+            std::cout << "Reached max polls" << std::endl;
+            return ::util::OkStatus();
+        }
+    }
+
+    auto status = reader->Finish();
+
+    // RPC failed
+    if (!status.ok()) {
+      return MAKE_ERROR(ERR_INTERNAL)
+        << "Subscribe rpc failed: "
+        << status.error_message();
+    }
+
+    std::cout << "Subscribe finished" << std::endl;
+
+    return ::util::OkStatus();
+  }
+
+  // Sets an attribute given the path into the PHAL attribute database.
+  ::util::Status HandleSet(const std::string& query) {
+    SetRequest req;
+    SetResponse resp;
+    grpc::ClientContext context;
+
+    // Create Set request
+    auto update = req.add_updates();
+    update->set_str(query);
+
+    // Grab value and set it based on type
+    while (true) {
+        // Grab value
+        std::string val_str;
+        std::cout << "Value: ";
+        std::getline(std::cin, val_str);
+        if (val_str == "") {
+            std::cout << "Invalid value." << std::endl;
+            continue;
+        }
+        // Grab type and then set it in the request
+        std::string type;
+        std::cout << "Type: ";
+        std::getline(std::cin, type);
+
+        if (!strcmp(type.c_str(), "int32")) {
+            ::google::protobuf::int32 val;
```

Comment:
What's the reason for using protobuf's integer definitions?

Reply:
should make no difference, happy to change to int32.